### DISCUSSION TITLE
How to setup custom validation for incubator.TextField

### DISCUSSION THREAD

**Discussion by navxio** [wix/react-native-ui-lib#1830] [Q&A] [answered]
Tried this but it doesn't work.. and there's no documentation either

`validate={['required', (data) => data.length >= 7]}`

**Comment by ethanshar** [comment]
Have you passed the `validationMessage` prop? 
and you should also pass `enableErrors` (it's false by default) 
If it's still not working, please share a full snippet of the code

**Comment by navxio** [comment]
@ethanshar I want to validate the length of the input, the mentioned code snippet pertains to the same but doesn't help.
Yes I have set the `validationMessage` prop along with `enableErrors`

Here's my full code:
```
      <TextField
        label="Password"
        placeholder="Enter your password"
        validate={['required', input => input.length > 7]}
        enableErros={true}
        secureTextEntry={true}
        autoComplete="password"
        validationMessage={'Password field cannot be empty'}
        validationMessagePosition="bottom"
        enableErrors={true}
        returnKeyType="next"
        returnKeyLabel="Next"
        onChangeText={(pwd) => setPassword(pwd)}
      />
```
So I want to force the users to use a password longer than 7 characters...

**Comment by ethanshar** [comment, marked as answer]
Hi @navxio 
You are using the old implementation of TextField, you can follow the warnings you get to migrate to the new implementation. 

Anyway, since we're in the middle of migration, this might be a little confusing. 
You have 2 options: 
- Pass the `migrate` prop. This will get you the new TextField implementation. 
- Use `Incubator.TextField` directly (e.g `<Incubator.TextField ... .>`)

Regardless, notice that in order to a separate validate message for each validation you can pass an array of validate messages (: 

good luck

**Reply by navxio** [reply]
so my validation for input length is implemented correctly?
